packages/core/src/modules/question-answer/QuestionAnswerModule.ts

```diff
@@ -0,0 +1,99 @@
+import type { ValidResponse } from './models'
+
+import { Lifecycle, scoped } from 'tsyringe'
+
+import { Dispatcher } from '../../agent/Dispatcher'
+import { MessageSender } from '../../agent/MessageSender'
+import { createOutboundMessage } from '../../agent/helpers'
+import { AriesFrameworkError } from '../../error'
+import { ConnectionService } from '../connections'
+
+import { AnswerMessageHandler, QuestionMessageHandler } from './handlers'
+import { QuestionAnswerState } from './models'
+import { QuestionAnswerService } from './services'
+
+@scoped(Lifecycle.ContainerScoped)
+export class QuestionAnswerModule {
+  private questionAnswerService: QuestionAnswerService
+  private messageSender: MessageSender
+  private connectionService: ConnectionService
+
+  public constructor(
+    dispatcher: Dispatcher,
+    questionAnswerService: QuestionAnswerService,
+    messageSender: MessageSender,
+    connectionService: ConnectionService
+  ) {
+    this.questionAnswerService = questionAnswerService
+    this.messageSender = messageSender
+    this.connectionService = connectionService
+    this.registerHandlers(dispatcher)
+  }
+
+  /**
+   * Create a question message with possible valid responses, then send message to the
+   * holder
+   *
+   * @param connectionId connection to send the question message to
+   * @param question question text included in message
+   * @param validResponses array of possible responses to question
+   * @param detail optional details for question
+   * @returns QuestionAnswer record
+   */
+  public async sendQuestion(connectionId: string, question: string, validResponses: ValidResponse[], detail?: string) {
```

Comment:
With this many parameters I'd lean to an object. In most modules we take the connection/record id as the first parameter and all other properties go into a second config parameter